src/Compilers/CSharp/Test/Semantic/Semantics/LocalFunctionTests.cs

```diff
@@ -745,6 +745,243 @@ void M()
+                //         [DllImport("a")] extern void local2() { } // 3, 4
+                Diagnostic(ErrorCode.ERR_ExternHasBody, "local2").WithArguments("local2()").WithLocation(11, 38),
+                // (12,10): error CS0601: The DllImport attribute must be specified on a method marked 'static' and 'extern'
+                //         [DllImport("a")] extern int local3() => 0; // 5, 6
+                Diagnostic(ErrorCode.ERR_DllImportOnInvalidMethod, "DllImport").WithLocation(12, 10),
+                // (12,37): error CS0179: 'local3()' cannot be extern and declare a body
+                //         [DllImport("a")] extern int local3() => 0; // 5, 6
+                Diagnostic(ErrorCode.ERR_ExternHasBody, "local3").WithArguments("local3()").WithLocation(12, 37),
+                // (14,21): error CS8112: Local function 'local4()' must either have a body or be marked 'static extern'.
+                //         static void local4(); // 7
+                Diagnostic(ErrorCode.ERR_LocalFunctionMissingBody, "local4").WithArguments("local4()").WithLocation(14, 21),
+                // (19,45): error CS0179: 'local8()' cannot be extern and declare a body
+                //         [DllImport("a")] static extern void local8() { } // 8
+                Diagnostic(ErrorCode.ERR_ExternHasBody, "local8").WithArguments("local8()").WithLocation(19, 45),
+                // (20,44): error CS0179: 'local9()' cannot be extern and declare a body
+                //         [DllImport("a")] static extern int local9() => 0; // 9
+                Diagnostic(ErrorCode.ERR_ExternHasBody, "local9").WithArguments("local9()").WithLocation(20, 44),
+                // (23,45): error CS0179: 'local11()' cannot be extern and declare a body
+                //         [DllImport("a")] extern static void local11() { } // 10
+                Diagnostic(ErrorCode.ERR_ExternHasBody, "local11").WithArguments("local11()").WithLocation(23, 45),
+                // (24,44): error CS0179: 'local12()' cannot be extern and declare a body
+                //         [DllImport("a")] extern static int local12() => 0; // 11
+                Diagnostic(ErrorCode.ERR_ExternHasBody, "local12").WithArguments("local12()").WithLocation(24, 44));
+        }
+
+        [Fact]
+        public void LocalFunctionExtern_Generic()
+        {
+            var source = @"
+using System.Runtime.InteropServices;
+
+class C1
+{
+#pragma warning disable 8321 // Unreferenced local function
+
+    void M1()
+    {
+        [DllImport(""a"")] extern static void local1();
+        [DllImport(""a"")] extern static void local2<T>(); // 1
+    }
+
+    void M2<T>()
+    {
+        [DllImport(""a"")] extern static void local3<T2>(); // 2
+    }
+}
+
+class C2<T>
+{
+    void M3()
+    {
+        [DllImport(""a"")] extern static void local4(); // 3
+        [DllImport(""a"")] extern static void local5<T2>(); // 4
+    }
+
+    void M4<T1>()
```

Comment:
>void M4<T1>() [](start = 4, length = 13)

Consider testing this scenario in a non-generic type. Also, scenario with local functions nested into local functions and lambdas, with generics on different levels.